roles/calico_master/tasks/main.yml

```diff
@@ -37,6 +37,35 @@
   set_fact:
     node_version: "{{ calico_node_image | regex_replace('^.*node:v?(.*)$', '\\1') }}"
     cnx: "{{ calico_node_image | regex_replace('^.*/(.*)-node:.*$', '\\1') }}"
+    use_calico_credentials: "{{ calico_image_credentials is defined | bool }}"
+
+- name: Calico | Encode Docker Credentials
+  shell: >
+    cat {{ calico_image_credentials }} | openssl base64 -A
+  register: calico_encoded_credentials_output
+  failed_when: "calico_encoded_credentials_output.rc != 0 or calico_encoded_credentials_output.stdout == ''"
+  when: use_calico_credentials
+
+- name: Calico | Set Encoded Docker Credentials Fact
+  set_fact:
+    calico_encoded_credentials: "{{ calico_encoded_credentials_output.stdout }}"
+  when: use_calico_credentials
+
+- name: Calico | Write Calico Pull Secret
+  template:
+    dest: "{{ mktemp.stdout }}/calico-pull-secret.yml"
+    src: calico-pull-secret.yml.j2
+  when: use_calico_credentials
+
+- name: Calico | Create Calico Pull Secret
+  when: use_calico_credentials
+  command: >
+    {{ openshift_client_binary }} apply
+    -f {{ mktemp.stdout }}/calico-pull-secret.yml
+    --config={{ openshift.common.config_base }}/master/admin.kubeconfig
+  register: calico_pull_secret_create_output
+  failed_when: "calico_pull_secret_create_output.rc != 0"
```

Comment:
This should exclude cases where stdout has `Already exists` - so that the playbook could be rerun several times and work properly on multi-master setup

Reply:
I might be misunderstanding, but I thought the apply command has an exit code of 0 when it modifies a resource, creates a resource, or leaves a resource unchanged?

Here are the results of testing it in my own environment:
```
$ oc apply -f test-ps.yaml
secret "test-calico-pull-secret" created
$ echo $?
0
$ oc apply -f test-ps.yaml
secret "test-calico-pull-secret" unchanged
$ echo $?
0
$ vim test-ps.yaml
$ oc apply -f test-ps.yaml
secret "test-calico-pull-secret" configured
$ echo $?
0
```

Am I looking at the right values?

Reply:
Ah, right, we're using `apply` here, it would error only when `create` is used